### PR TITLE
[clangd] Avoid clangd crash with -fsycl flag

### PR DESCRIPTION
**Pull request by rafbiels**
Add four extra option resets in `disableUnsupportedOptions` when parsing SYCL code. All four are needed to make clangd happily parse code when `-fsycl` is specified in compile commands. Without each of them, the following happens:
* without `SYCLIsDevice=false`: crash on !nullptr assert
* without `DeclareSPIRVBuiltins=false`: crash on !nullptr assert
* without `SYCLUnnamedLambda=false`: `ref_non_value` error, `'T' does not refer to a value`
* without `Triple=HostTriple`: `pp_file_not_found` error, `'gnu/stubs-32.h' file not found`

Fixes intel/llvm#11088 and clangd/clangd#1097

Also add a LIT test which parses the following code:
```cpp
#include <sycl/sycl.hpp>
sycl::queue q{};
```
and checks `symbolInfo` for the symbol `q` against a reference.

### CODE REVIEW

**Comment by premanandrao** [comment]
Not sure who the reviewers for this should be.

**Comment by elizabethandrews** [comment]
> Not sure who the reviewers for this should be.

Yea me neither. @bader the FE team does not have experience with clangd. Who should the code owners be for this PR?

**Comment by bader** [comment]
> > Not sure who the reviewers for this should be.
> 
> Yea me neither. @bader the FE team does not have experience with clangd. Who should the code owners be for this PR?

I don't know anyone who works with clangd on this project. Considering that this is a clang tool @intel/dpcpp-cfe-reviewers is the best option we have. Based on the changes I guess @intel/dpcpp-clang-driver-reviewers team might be able to help, but I don't know if clangd shares any logic with the clang driver.

File: clang-tools-extra/clangd/Compiler.cpp [hunk 1]
```diff
@@ -89,6 +89,12 @@ void disableUnsupportedOptions(CompilerInvocation &CI) {
   CI.getLangOpts().ProfileListFiles.clear();
   CI.getLangOpts().XRayAlwaysInstrumentFiles.clear();
   CI.getLangOpts().XRayNeverInstrumentFiles.clear();
+  if (CI.getLangOpts().SYCLIsDevice) {
+    CI.getLangOpts().SYCLIsDevice = false;
+    CI.getLangOpts().SYCLUnnamedLambda = false;
+    CI.getLangOpts().DeclareSPIRVBuiltins = false;
+    CI.getTargetOpts().Triple = CI.getTargetOpts().HostTriple;
```
Comment (premanandrao) [inline clang-tools-extra/clangd/Compiler.cpp:96]: Are there test cases you can add for these changes?

Comment (rafbiels) [reply clang-tools-extra/clangd/Compiler.cpp:96]: Very good point @premanandrao, I agree this should be tested. Added a LIT test parsing basic SYCL code. The test reproduces the crash without these changes and succeeds with them.

**Comment by rafbiels** [comment]
@intel/dpcpp-cfe-reviewers I would appreciate any further review comments or launching the tests if this version looks OK.

**Review by premanandrao — APPROVED** [review]
LGTM

**Pull request merged**